service/src/test/java/org/databiosphere/workspacedataservice/controller/JobControllerTest.java

```diff
@@ -0,0 +1,115 @@
+package org.databiosphere.workspacedataservice.controller;
+
+import static org.databiosphere.workspacedataservice.generated.GenericJobServerModel.JobTypeEnum;
+import static org.databiosphere.workspacedataservice.generated.GenericJobServerModel.StatusEnum;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.List;
+import java.util.UUID;
+import org.databiosphere.workspacedataservice.dao.JobDao;
+import org.databiosphere.workspacedataservice.generated.GenericJobServerModel;
+import org.databiosphere.workspacedataservice.shared.model.InstanceId;
+import org.databiosphere.workspacedataservice.shared.model.job.Job;
+import org.databiosphere.workspacedataservice.shared.model.job.JobInput;
+import org.databiosphere.workspacedataservice.shared.model.job.JobResult;
+import org.databiosphere.workspacedataservice.shared.model.job.JobType;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.test.annotation.DirtiesContext;
+
+@DirtiesContext
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class JobControllerTest {
+  @Autowired private NamedParameterJdbcTemplate namedTemplate;
+
+  @Autowired private TestRestTemplate restTemplate;
+
+  @Autowired private JobDao jobDao;
+
+  private InstanceId instanceId = InstanceId.of(UUID.randomUUID());
+
+  private UUID jobId;
+
+  @BeforeAll
+  void beforeAll() {
+    // push jobs into the db, one of different instanceId
+    Job<JobInput, JobResult> testJob1 =
+        Job.newJob(instanceId, JobType.DATA_IMPORT, JobInput.empty());
```

Comment:
nit: there's a lot of `Job` showing up in the test setup...most of them are fine, but I think the `Job.newJob` would benefit from using a static import to just become:

```suggestion
       newJob(instanceId, JobType.DATA_IMPORT, JobInput.empty());
```

Would require:
```
import static org.databiosphere.workspacedataservice.shared.model.job.Job.newJob;
```